### DISCUSSION TITLE
What do you think of using a longer names for generic type variables? (Expect<SubjectT> instead of Expect<T>)

### DISCUSSION THREAD

**Discussion by vlsi** [robstoll/atrium#1700] [Contributor Q&A] [answered]
See https://google.github.io/styleguide/javaguide.html#s5.2.8-type-variable-names

WDYT?

E.g. 

```kotlin
infix fun <T, R> Expect<T>.feature(property: KProperty1<in T, R>): FeatureExpect<T, R> =

// =>

infix fun <SubjectT, PropertyT> Expect<SubjectT>.feature(property: KProperty1<in SubjectT, PropertyT>): FeatureExpect<SubjectT, PropertyT> =
```

**Comment by robstoll** [comment]
the style guide mentions also the short form 😁. just kidding, I am with you, we already use longer names in some cases ( though we use TSubject instead of SubjectT but both fine with me) I don't mind if we are clear in all cases. 

I'll create an issue once I am in front of a Screen

**Reply by vlsi** [reply]
Just to clarify, a case with a single T is probably good enough.

The following might probably benefit from longer names:

```kotlin
class FailureHandlerAdapter<T, R1, R>(
    val failureHandler: SubjectChanger.FailureHandler<R1, R>,
    val map: (T) -> R1
) : SubjectChanger.FailureHandler<T, R> {
```

**Reply by robstoll** [reply, marked as answer]
So let's do it on a case by case basis. I won't create an issue therefore and instead, contributors can feel free to change the naming in PRs if they should be affected. I think it would make sense to document the type parameters in case of a single T so that it is still clear (sometimes we do it but I think not always)
